### DISCUSSION TITLE
How to make a Popup with CornerRadius?

### DISCUSSION THREAD

**Discussion by josephnarai** [AvaloniaUI/Avalonia#6716] [Q&A] [answered]
I want to make a simple Popup with rounded corners, so as per the Border example here:  

https://docs.avaloniaui.net/docs/controls/border

I've put that border inside the Popup.

```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaButtonTest.MainWindow"
        Title="AvaloniaButtonTest">
  <StackPanel Name="MyStackPanel">
    <Popup IsOpen="True">
      <Border Background="Red" Width="200" BorderBrush="Black" BorderThickness="2" CornerRadius="20" Padding="4">
        <StackPanel>
          <Button>Button 1</Button>
          <Button>Button 2</Button>
        </StackPanel>
      </Border>
    </Popup>
  </StackPanel>
</Window>
```

However, the Popup seems to have it's own non rounded border with shadow.

![image](https://user-images.githubusercontent.com/14188194/137222939-673c6cb6-aafc-4ff1-a755-452627d6b80c.png)

Any ideas how to get rid of the Popup shadow and border?

**Comment by maxkatz6** [comment, marked as answer]
You need to disable WindowManagerAddShadowHint
See: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml#L122

**Reply by maxkatz6** [reply]
It makes sense to add it to the documentation, but probably we are good to have it disabled by default...

**Reply by josephnarai** [reply]
Thanks, I was looking for popup axaml... Didn't expect it to be in combobox!

Is there a way to disable in code? Or only via axaml?

**Reply by maxkatz6** [reply]
ComboBox was just as an example of this property usage.
By itself it's defined here https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Primitives/Popup.cs#L26

**Reply by maxkatz6** [reply]
Sure, just set WindowManagerAddShadowHint property to false

**Reply by josephnarai** [reply]
Awesome, thanks!

**Comment by maxkatz6** [comment]
If anybody will find this discussion in the future. Related:
https://github.com/AvaloniaUI/Avalonia/pull/6717
